src/js/components/FrameworkConfiguration.js

```diff
@@ -198,34 +198,55 @@ class FrameworkConfiguration extends Component {
     return actions;
   }
 
-  getReviewScreen() {
-    const {
-      packageDetails,
-      deployErrors,
-      isInitialDeploy,
-      formData
-    } = this.props;
+  getTermsAndConditions() {
+    const { packageDetails } = this.props;
 
-    const fileName = "config.json";
-    const configString = JSON.stringify(formData, null, 2);
+    const termsUrl = packageDetails.isCertified()
+      ? "https://mesosphere.com/catalog-terms-conditions/#certified-services"
+      : "https://mesosphere.com/catalog-terms-conditions/#community-services";
 
-    const renderKeys = this.getHashMapRenderKeys(formData);
+    return {
+      __html: `By running this service you agree to the <a href=${termsUrl} target='_blank' title='Terms and Conditions'>terms and conditions</a>.`
+    };
+  }
+
+  getWarningMessage() {
+    const { packageDetails } = this.props;
 
     const preInstallNotes = packageDetails.getPreInstallNotes();
-    let preinstall = null;
-    if (preInstallNotes && isInitialDeploy) {
-      const preInstallNotesParsed = StringUtil.parseMarkdown(preInstallNotes);
-      preInstallNotesParsed.__html =
-        "<strong>Preinstall Notes: </strong>" + preInstallNotesParsed.__html;
+    if (preInstallNotes) {
+      const message = StringUtil.parseMarkdown(preInstallNotes);
+      message.__html = `<strong>Preinstall Notes: </strong>${message.__html} ${this.getTermsAndConditions().__html}`;
 
-      preinstall = (
+      return (
         <div
-          dangerouslySetInnerHTML={preInstallNotesParsed}
+          dangerouslySetInnerHTML={message}
```

Comment:
any chance to get rid of this?

Reply:
I think the answer is "no" since those messages can contain html we want to render.

Reply:
Yeah, this is a pattern we follow [elsewhere](https://github.com/dcos/dcos-ui/blob/master/src/js/pages/catalog/PackageDetailTab.js#L429) after parsing markdown. It's ok - I don't feel too dangerous doing this 😉